lib/server.js

```diff
@@ -204,11 +215,26 @@ class Server {
     };
 
     if (existingConnection) {
-      getSessionCallback(existingConnection.session);
-    } else if (sessionStorageProvider.isAsync()) {
-      sessionStorageProvider.get(sessionId, getSessionCallback);
+      const replacementTransport = connection.transport;
+      Object.keys(connection._transportListeners).forEach((event) => {
+        connection.transport.removeListener(
+          event, connection._transportListeners[event]
+        );
+      });
+      connection.transport = null;
```

Comment:
I think it's better to move this (218-224) to Connection to something like `_removeTransport()` that will return removed transport?

Reply:
Yeah, sure.